src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -4,22 +4,467 @@ Thomas Segismont
 :icons: font
 :jbake-type: page
 :jbake-status: published
+:toc: macro
+:toc-title:
 
-Hawkular Metrics is the metric data store part of the http://www.hawkular.org/[Hawkular] organization.
-It is currently used as the metric storage engine for Hawkular but it can also be used independently. The project relies on https://cassandra.apache.org/[Apache Cassandra] as a backend.
+toc::[]
 
+== Introduction
+Hawkular Metrics is a metrics storage engine built on top of link:http://cassandra.apache.org[Cassandra] that offers
+a rich, growing set of features that includes:
 
-Three types of metrics are currently supported:
+* Multi tenancy
+* Multiple metric types
+* Tagging
+* Automatic data removal
+* REST API
+* Scalability
+* Rich query support
 
- * Availability: metric that represent availability: up or down
- * Counter: numeric value that always goes up; example: the total number of visitors of a web page
-   ** Rates: rate of change can be retrieve for counter rates
- * Gauge: numeric value that can go up or down; example: the response time on an application
-   ** Aggregate buckets (min/max/average on some intervals) can be obtained for gauges when individual datapoints are not needed
+== Identifiers
+All identifiers are stored as variable length, UTF-8 encoded strings. This includes:
 
-Tags are available for easy filtering and querying, along with retention times for automatic pruning of older data.
+* Tenant ids
+* Metric names
+* Tag keys (for both metric and data point tags)
 
+[TIP]
+At present there is no restriction on characters that can be used in identifiers. This may change in the future
+though (See link:https://issues.jboss.org/browse/HWKMETRICS-208[HWKMETRICS-208] for details). For this reason it is
+recommended to restrict the characters to letters, numbers, underscore, period, and forward slash.
```

Comment:
What should be mentioned here is that in the URL all the special characters defined in the HTTP standard must be encoded. This is clearly not obvious to many people. Forward slash is no exception to this rule although many applications are badly coded to handle it.